mdata/cache/ccache_metric.go

```diff
@@ -116,12 +231,29 @@ func (mc *CCacheMetric) Add(prev uint32, itergen chunk.IterGen) {
 		}
 	}
 
-	// regenerate the list of sorted keys after adding a chunk
-	mc.generateKeys()
+	mc.addKey(ts)
 
 	return
 }
 
+func (mc *CCacheMetric) addKey(ts uint32) {
+
+	// if no keys yet, just add it and it's sorted
+	if len(mc.keys) == 0 {
+		mc.keys = append(mc.keys, ts)
+		return
+	}
+
+	// if ts is newer than any previous chunk, can just add to the back
+	if mc.keys[len(mc.keys)-1] < ts {
+		mc.keys = append(mc.keys, ts)
+		return
+	}
+
+	// we have to insert it in the middle, needs to be re-generated
```

Comment:
in this case, where did the key actually get inserted?

Reply:
so instead of the above 
```
if mc.keys[len(mc.keys)-1] < ts {
	mc.keys = append(mc.keys, ts)
	return
}
```
it could just be
```
mc.keys = append(mc.keys, ts)
if mc.keys[len(mc.keys)-1] < mc.keys[len(mc.keys)-2] {
    mc.generateKeys()
}
```